lib/collection/src/shards/local_shard/updaters.rs

```diff
@@ -0,0 +1,85 @@
+use std::sync::Arc;
+
+use tokio::sync::mpsc;
+
+use crate::operations::types::CollectionResult;
+use crate::optimizers_builder::build_optimizers;
+use crate::shards::local_shard::LocalShard;
+use crate::update_handler::UpdateSignal;
+
+impl LocalShard {
+    pub fn trigger_optimizers(&self) {
+        // Send a trigger signal and ignore errors because all error cases are acceptable:
+        // - If receiver is already dead - we do not care
+        // - If channel is full - optimization will be triggered by some other signal
+        let _ = self.update_sender.load().try_send(UpdateSignal::Nop);
+    }
+
+    pub async fn stop_flush_worker(&self) {
+        let mut update_handler = self.update_handler.lock().await;
+        update_handler.stop_flush_worker()
+    }
+
+    pub async fn wait_update_workers_stop(&self) -> CollectionResult<()> {
+        let mut update_handler = self.update_handler.lock().await;
+        update_handler.wait_workers_stops().await
+    }
+
+    /// Synchronously asks update workers to stop
+    ///
+    /// This function doesn't wait for workers to actually stop, but it should finish quickly.
+    /// Returns true if the workers were already stopped.
+    /// Returns false if workers might still be running.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if called from async runtime.
+    /// That includes sync functions called from async context.
+    pub fn blocking_ask_workers_to_stop(&self) -> bool {
+        {
+            let mut update_handler = self.update_handler.blocking_lock();
+            if update_handler.is_stopped() {
+                return true;
+            }
+            update_handler.stop_flush_worker();
+        }
+
+        // This can block longer, if the channel is full
+        // If channel is closed, assume it is already stopped
+        if let Err(err) = self.update_sender.load().blocking_send(UpdateSignal::Stop) {
+            log::trace!("Error sending update signal to update handler: {err}");
+        }
+
+        false
+    }
+
+    pub async fn on_optimizer_config_update(&self) -> CollectionResult<()> {
```

Comment:
This is **not** cancel safe. We must annotate that, and ensure we don't cancel it in any way.